### DISCUSSION TITLE
Quotes are automatically added when using the home directory

### DISCUSSION THREAD

**Discussion by xicheng412** [google/zx#377] [Help needed]
when I use the following code

```js
const home="~"
$`ls ${home}`
```

### Expected Behavior

shell run the command `> ls ~`

### Actual Behavior

home directory changed `> ls $'~'`

### reason

In the file `src/index.mjs`

```
function quote(arg) {
  if (/^[a-z0-9/_.-]+$/i.test(arg) || arg === '') {
    return arg
  }
....
}
```
the filter dose not contain ~.

How should I write it when I need to use a path that includes the home directory?

**Comment by antonmedv** [comment]
Well, this is little bit more complicated question. 

First, please, read https://github.com/google/zx/blob/main/docs/quotes.md. 

Right now, no substitution is made for ~ tilda. We can take a look on how it’s implemented right now in bash and do the same. 

Problem with allowing tilda: how are you going to do this:

    echo “~”

**Reply by xicheng412** [reply]
I appreciate your support very much. Actually I am not using the Home directory locally, I'm running script like:

```bash
ssh server1 "sh ~/cmd.sh"
```

I have some different servers with different username. ssh config contains the infomation about username. `os.homedir()` can not use in remote server.

I solved the problem with this method.

```js
const {stdout:pwdRes} = await $`ssh ${serverName} "pwd"`
const remoteHome = pwdRes.split('\n')[0]
console.log(path.join(remoteHome,remote.base,remote.file))
```

While default login path is home dirctory, I saved the path as remoteHome, then use `path.join` to get full path.

*The following is very important and useful information in the documentation:*

### globbing and `~`

As everything passed through `${...}` will be escaped, you can't use `~` or glob
syntax. 

In order for this to work the zx provides 
[globby package](../README.md#globby-package).

For instead of this:

```js
let files = '~/dev/**/*.md' // wrong
await $`ls ${files}`
```

Use `glob` function and `os` package:

```js
let files = await glob(os.homedir() + '/dev/**/*.md')
await $`ls ${files}`
```
